src/utils/hooks.ts

```diff
@@ -1,16 +1,95 @@
-import { graphql, useStaticQuery } from 'gatsby'
-
-export const useSiteMetadata = () => {
-  const { site } = useStaticQuery(
-    graphql`
-      query {
-        site {
-          siteMetadata {
-            title
-          }
-        }
-      }
-    `,
-  )
-  return site.siteMetadata
+import { useCallback, useEffect, useLayoutEffect, useState } from 'react'
+
+export const useDimensions = () => {
+  const [dimensions, setDimensions] = useState({ width: 0 })
+  const [node, setNode] = useState(null)
+
+  const ref = useCallback(node => {
+    setNode(node)
+  }, [])
+
+  useLayoutEffect(() => {
+    if (node == null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 spaces but found 4. (@typescript-eslint/indent)](https://app.codacy.com/app/jdstuart/jesses.io/pullRequest?prid=4576429)